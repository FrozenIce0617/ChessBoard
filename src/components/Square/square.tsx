import React, { ReactNode } from 'react';

interface IProps {
  isBlack: boolean;
  isOverlay: boolean;
  children?: ReactNode;
}

class Square extends React.Component<IProps> {
  render() {
    const { isBlack, isOverlay, children } = this.props;
    const fill = isBlack ? 'black' : 'white';
    const stroke = isBlack ? 'white' : 'black';

    return (
      <div
        style={{
          backgroundColor: isOverlay ? 'red' : fill,
          opacity: isOverlay ? 0.5 : 1,
          color: stroke,
          width: '100%',
          height: '100%',
          userSelect: 'none',
        }}
      >
        {children}
      </div>
    );
  }
}

export default Square;
