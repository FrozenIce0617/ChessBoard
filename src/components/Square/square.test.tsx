import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Square from './square';

describe('Square Component', () => {
  it('render component', () => {
    const component = shallow(<Square isBlack={true} isOverlay={false} />);

    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
