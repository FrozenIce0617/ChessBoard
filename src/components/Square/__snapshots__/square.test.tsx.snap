// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square Component render component 1`] = `
<div
  className="square"
  style={
    Object {
      "backgroundColor": "black",
      "color": "white",
      "opacity": 1,
    }
  }
/>
`;
